allennlp/common/testing/test_case.py

```diff
@@ -18,8 +18,13 @@ class AllenNlpTestCase:
     PROJECT_ROOT = (pathlib.Path(__file__).parent / ".." / ".." / "..").resolve()
     MODULE_ROOT = PROJECT_ROOT / "allennlp"
     TOOLS_ROOT = MODULE_ROOT / "tools"
-    TESTS_ROOT = PROJECT_ROOT / "tests"
-    FIXTURES_ROOT = PROJECT_ROOT / "test_fixtures"
+    # to run test suite with finished package, which does not contain
+    # tests & fixtures, we must be able to look them up somewhere else
+    PROJECT_ROOT_FALLBACK = (
+        pathlib.Path(os.environ["SRC_DIR"]) if "SRC_DIR" in os.environ else PROJECT_ROOT
```

Comment:
This is the current state of what I'm using on the feedstock, but if people are concerned that other users might (by accident) have an environment variable `SRC_DIR`, then the if-condition could easily be replaced by something less ambiguous (see e.g. [here](https://docs.conda.io/projects/conda-build/en/latest/user-guide/environment-variables.html?highlight=cmake#id2))
```suggestion
        pathlib.Path(os.environ["SRC_DIR"]) if "CONDA_BUILD" in os.environ else PROJECT_ROOT
```

OTOH, I'm not sure how many users want to run the test suite of the installed package (where they'd run into the same problems as I do), but by being able to set `SRC_DIR`, they'd actually have a chance to succeed.

Reply:
The `SRC_DIR` does sound a little too generic. But how about `ALLENNLP_SRC_DIR`?

Reply:
`SRC_DIR` is what conda build uses already (so using the presence of `CONDA_BUILD`, this would be the most efficient for me), but indeed, I could also set `ALLENNLP_SRC_DIR`.

Would you consider a two-part fall-back?
```
    PROJECT_ROOT_FALLBACK = (
        # users wanting to run test suite for installed package
        pathlib.Path(os.environ["ALLENNLP_SRC_DIR"])
        if "ALLENNLP_SRC_DIR" in os.environ
        else (
            # fallback for conda packaging
            pathlib.Path(os.environ["SRC_DIR"])
            if "CONDA_BUILD" in os.environ
            # stay in-tree
            else PROJECT_ROOT
        )
    )
```

That way, both cases would be covered - general users could set `ALLENNLP_SRC_DIR`, and I wouldn't have to set environment variables all over the place (e.g. if we took the same kind of approach for tango, I'd have to set it 6 times, one for each time some part of the test suite gets run).

Reply:
I like that 🙂